### PR TITLE
Add prevten validations

### PR DESCRIPTION
**Pull request by kosiakkatrina**
prevten cannot be 13 (children's home/foster care) if lead tenant is 20 or older
prevten cannot be 21 (refuge) if person 1 is a man
prevten cannot be any of the following if the answer to Q16 `What was the source of referral for this letting?` is 1 Internal Transfer 3, 4, 10, 13, 19, 23, 24, 25, 26, 28, 29

### CODE REVIEW

**Review by dushan-madetech — APPROVED** [review]
Didn't find any 🧦 but otherwise looks fine to me. 🚢 it

File: app/models/validations/household_validations.rb [hunk 1]
```diff
@@ -64,6 +64,22 @@ def validate_previous_housing_situation(record)
     if record.is_relet_to_temp_tenant? && !record.previous_tenancy_was_temporary?
       record.errors.add :prevten, I18n.t("validations.household.prevten.non_temp_accommodation")
     end
+
+    if record.age1.present? && record.age1 > 19 && record.prevten == 13
```
Comment (baarkerlounger) [inline app/models/validations/household_validations.rb]: Possibly more readable to add a wrapper method like `previous_tenancy_was_foster_care?` to `case_log`?

File: spec/models/validations/household_validations_spec.rb [hunk 2]
```diff
@@ -578,5 +578,51 @@
...
+        expect(record.errors["referral"])
+        .to include(match I18n.t("validations.household.referral.prevten_invalid", prevten: ""))
+      end
+
+      it "cannot be 4, 10, 13, 19, 23, 24, 25, 26, 28, 29" do
+        record.referral = 1
+        record.prevten = 4
+        household_validator.validate_previous_housing_situation(record)
+        expect(record.errors["prevten"])
+        .to include(match I18n.t("validations.household.prevten.internal_transfer", prevten: ""))
```
Comment (baarkerlounger) [inline spec/models/validations/household_validations_spec.rb]: Indentation here doesn't match above

**Review by baarkerlounger — APPROVED** [review]

File: app/models/validations/household_validations.rb [hunk 3]
```diff
@@ -64,6 +64,22 @@ def validate_previous_housing_situation(record)
     if record.is_relet_to_temp_tenant? && !record.previous_tenancy_was_temporary?
       record.errors.add :prevten, I18n.t("validations.household.prevten.non_temp_accommodation")
     end
+
+    if record.age1.present? && record.age1 > 19 && record.previous_tenancy_was_foster_care?
+      record.errors.add :prevten, I18n.t("validations.household.prevten.over_20_foster_care")
+      record.errors.add :age1, I18n.t("validations.household.age.lead.over_20")
+    end
+
+    if record.sex1 == "M" && record.prevten == 21
```
Comment (baarkerlounger) [inline app/models/validations/household_validations.rb]: Could do the same for these two but also happy to merge this

**Pull request merged**